language/move-vm/runtime/src/interpreter.rs

```diff
@@ -124,13 +125,22 @@ impl Interpreter {
                 .map_err(|e| self.set_location(e))?;
         }
 
+        let func_call_stack = self
+            .call_stack
+            .to_func_call_stack(function.name())
+            .map_err(|e| self.set_location(e))?;
```

Comment:
I think there are some issues with the implementation here. You only compute the call stack once at the entry point and don't recompute/modify it when a call is made, which means the stack will only contain the entry point ever.

Reply:
Yup you're right, this should be inside the loop - got mixed up while rebasing